apps/web/src/stores/documentVersions.ts

```diff
@@ -19,30 +24,37 @@ export default function useDocumentVersions(
     new URLSearchParams({
       staged: String(staged),
     }).toString()
-  const { mutate, data, ...rest } = useSWR(
+
+  const { mutate, data, ...rest } = useSWR<DocumentVersion[]>(
     key,
     (url: string) => fetch(url).then((res) => res.json()),
     opts,
   )
-  const create = async (payload: {
-    commitUuid?: string
-    name: string
-    documentType?: DocumentVersion['documentType']
-    parentId?: number
-  }) => {
-    try {
-      const doc = await createDocumentVersionAction({
+  const documents = data ?? []
+  const { execute } = useServerAction(createDocumentVersionAction)
+  const create = useCallback(
+    async (payload: {
+      commitUuid?: string
+      name: string
+      documentType?: DocumentType
+      parentId?: number
+    }) => {
+      const [document] = await execute({
         ...payload,
+        projectId: FIXME_HARDCODED_PROJECT_ID,
         name: payload.name!,
         commitUuid: payload.commitUuid || HEAD_COMMIT,
       })
-      mutate([...data, doc])
+      const prev = documents ?? []
 
-      return doc
-    } catch (err) {
-      console.error(err)
-    }
-  }
+      if (document) {
+        mutate([...prev, document])
```

Comment:
Instead of dealing with documents and then the UI having to do the `toTree` this end point could return the tree directly